Fix bug for entirely empty chunks of rows

jacobcook1995 opened the pull request:
This PR adds a check that the set of valid dates isn't empty before attempting to use this to update the temporal bounds. This comes from a weird edge where a chunk of rows contains entirely "NA" and so every single entry is ignored

I haven't added a test to demonstrate that this works, if you think of an appropriate one let me know.

jacobcook1995 committed:
Added check that data isn't empty

davidorme reviewed (approved):
LGTM

davidorme commented:
I guess a test would be to have an in-memory workbook with a couple of NA rows and then set the chunksize to 1 or 2 to keep it small?

jacobcook1995 committed:
Added test of NA handling for DatetimeField

jacobcook1995 committed:
Added NA tests for TimeField

jacobcook1995 commented:
Added a test for "NA" data entries for the DatetimeField. I had to change how the consistency checking is done as this was producing "Date or datetime data mixes ISO string and time formatted rows" errors when it shouldn't.

I also added a similar test + change to the consistency checking for the TimeField as this field uses a very similar logic. 

Considered adding similar tests for the other fields but that felt like a real rabbit hole for an edge case.

davidorme reviewed (approved):
LGTM.
